FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -34,11 +36,14 @@ public DefaultClientIO(Uri domain, TimeSpan timeout, string secret)
             client.DefaultRequestHeaders.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
             client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
 
-            authHeader = AuthString(secret);
+            authHeader = AuthHeader(secret);
         }
 
         public IClientIO NewSessionClient(string secret) =>
-            new DefaultClientIO(client, secret);
+            new DefaultClientIO(client, AuthHeader(secret), useRecursiveRefs);
+
+        public IClientIO NewWithRecursiveRefs(bool useRecursive) =>
```

Comment:
My suggestion would be to add the the API version to the constructor, probably as a constant, enum, or something like that. We should make 2.0 the default for now so the driver still works as it is and tests will pass.

That also makes it easy to setup the client as we make changes in the new api. We could set different parsers, for example.

The `NewSessionClient` should remain unchanged, IMO. That way, we make the user take the decision of which API version to use one place: the driver initialization only.